File: Easy/path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {        // 1ms
    public boolean recursion(TreeNode root, int targetSum) {
        if(root.left == null && root.right == null)
            return root.val == targetSum;
        
        if(root.left != null && recursion(root.left, targetSum - root.val))
            return true;
        
        if(root.right != null && recursion(root.right, targetSum - root.val))
            return true;
        
        return false;
    }
    public boolean hasPathSum(TreeNode root, int targetSum) {
        if(root == null) return false;
        return recursion(root, targetSum);
    }
}

File: Easy/path-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def recursion(self, node: Optional[TreeNode], targetSum: int) -> bool:
        if node.left == None and node.right == None:
            return node.val == targetSum
        left = False
        if(node.left != None):
            left = self.recursion(node.left, targetSum - node.val)
        right = False
        if(node.right != None):
            right = self.recursion(node.right, targetSum - node.val)
        return left or right
    
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if root == None:
            return False
        return self.recursion(root, targetSum)